packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/EmitterRpc/Emitter.cs

```diff
@@ -13,23 +13,21 @@ public sealed class Emitter : IDisposable
         private const string Trace = "trace";
         private const string Diagnostic = "diagnostic";
 
-        private static Emitter? _emitter;
         private bool _disposed;
 
         private readonly StreamWriter _writer;
 
-        private Emitter()
+        internal Emitter(Stream stream)
         {
-            _writer = new StreamWriter(Console.OpenStandardOutput()) { AutoFlush = true };
+            _writer = new StreamWriter(stream) { AutoFlush = true };
         }
 
-        public static Emitter Instance => _emitter ??= new Emitter();
```

Comment:
the singleton pattern was messing things up in our test cases because it might dispose the stream prematurely
therefore I decide to remove the singleton pattern here, and when we load the plugin, we inject the constructed instance in main into the `CodeModelPlugin`.